src/schemathesis/filters.py

```diff
@@ -17,3 +17,16 @@ def should_skip_endpoint(endpoint: str, pattern: Optional[Filter]) -> bool:
         return False
     patterns = force_tuple(pattern)
     return not any(re.search(item, endpoint) for item in patterns)
+
+
+def should_skip_by_tag(tags: Optional[List[str]], pattern: Optional[Filter]) -> bool:
+    if pattern is None:
+        return False
+    patterns = force_tuple(pattern)
+    matches = []
+    if tags:
```

Comment:
if there are no tags, then we know that it should be skipped - no tags to match. this check could be done just after the condition in the beginning of the function